catkin_ws/src/kpr_arm_control/src/arm_control/node_arm_control.cpp

```diff
@@ -0,0 +1,127 @@
+#include "ros/ros.h"
+#include "arm_control.cpp"
+#include <memory>
+
+#include "cucumber_msgs/HarvestAction.h"
+#include "cucumber_msgs/Cucumber.h"
+#include "cucumber_msgs/CucumberContainer.h"
+#include "geometry_msgs/Pose.h"
+#include "geometry_msgs/Quaternion.h"
+
+#include <ur_msgs/SetIO.h>
+
+#include <moveit/move_group_interface/move_group.h>
+#include <moveit/planning_scene_interface/planning_scene_interface.h>
+
+using namespace ros;
+
+const std::string NODE_NAME = "Arm Control";
+const std::string move_group_name("manipulator");
+
+/**
+ * Attempts to pick a cucumber. This function does the following things:
+ * - It attempts to move the arm to the location of the cucumber.
+ * - It attempts to grip the cucumber.
+ * - It attempts to cut the stem of the cucumber.
+ * - It attempts to move the arm to the drop-off location.
+ * - It attempts to let the cucumber go.
+ * 
+ * @param msg The Request to pick a certain cucumber.
+ * @param response The Response that will be sent to the client.
+ * 
+ * @return True if the service didn't encounter critical errors, false otherwise.
+ */
+bool getCucumber(cucumber_msgs::HarvestAction::Request &msg,
+		cucumber_msgs::HarvestAction::Response &response) {
+	bool success = true;
+
+	ROS_INFO("Got cucumber");
```

Comment:
Please change this ambiguous logging text. You also might want to change most of theses logs to the DEBUG level and only leave the first and last log on the INFO level